lib/serialization.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const { iter } = require('@metarhia/common');
+const { extractDecorator } = require('metaschema');
+
+const { stringifyInstance } = require('./utils');
+
+const {
+  serializeDecorated,
+  supportDecorators,
+} = require('./decorators-serialization');
+
+const replacer = (name, value) =>
+  typeof value === 'function' ? value.toString() : value;
+
+const serialize = (definition, excludedKeys, type) =>
+  iter(Object.entries(definition))
+    .map(([name, ins]) => {
+      const decName = extractDecorator(ins);
+
+      let stringifyValue;
+      if (decName === 'Object') {
+        stringifyValue = stringifyInstance(ins, excludedKeys, replacer);
+      } else if (supportDecorators[type].has(decName)) {
+        stringifyValue = serializeDecorated(type, decName, ins, excludedKeys);
+      } else {
+        return null;
```

Comment:
I'd prefer to check if we should serialize property earlier, maybe add a map that will extract decorator and return tuple `(decorator, property, value)`.